src/main/java/com/prolinkli/core/app/components/user/service/UserAuthService.java

```
@@ -63,26 +65,24 @@ public AuthorizedUser login(UserAuthenticationForm userAuthForm) {
 
   public AuthorizedUser refresh(AuthorizedUser user, HttpServletResponse response) {
     if (user == null) {
-      throw new IllegalStateException("User not authenticated");
+      throw new AuthorizationDeniedException("User not authenticated");
     }
 
     return jwtSaveService.regenerateTokens(user, response);
   }
 
-  private Map<String, Object> getCredentials(UserAuthenticationForm userAuthForm) {
+  private void getCredentials(UserAuthenticationForm userAuthForm) {
```

The `getCredentials` method could be improved:

1. **Missing validation**: No handling for unknown authentication methods - should throw an exception for unsupported auth types
2. **Parameter mapping**: Could be more robust with null checks and validation of required fields before mapping
3. **Architecture**: Consider using a strategy pattern instead of if/else chains - would make adding new auth methods much cleaner

Current approach assumes valid auth method and required fields, which could cause runtime issues.

---

this could be a good todo to start writing validator classes that use a strategy pattern to start checking things; good for sprint 2